File: scripts/python/verify.py
#!/usr/bin/env python3
"""verify.py — runs the local pre-push verification suite for a Flutter project.
verify.py — 运行 Flutter 项目推送前的本地校验流程。

Usage:  python scripts/python/verify.py [--no-publish]
用法：  python scripts/python/verify.py [--no-publish]
Runs:   flutter analyze, flutter test, and (unless skipped) flutter pub publish --dry-run.
执行：  flutter analyze、flutter test，以及（除非跳过）flutter pub publish --dry-run。
Exits non-zero if any step fails, so it can gate CI or a pre-push hook.
任一环节失败即以非零码退出，可用于 CI 或 pre-push 钩子拦截。
"""

import subprocess
import sys


def run(cmd, args):
    print("\n> {} {}".format(cmd, " ".join(args)))
    result = subprocess.run(cmd, args if isinstance(args, list) else list(args),
                            shell=(sys.platform == "win32"))
    return result.returncode


def main(argv):
    skip_publish = "--no-publish" in argv

    # 1. Static analysis.
    # 1. 静态分析。
    if run("flutter", ["analyze"]) != 0:
        sys.stderr.write("flutter analyze failed.\n")
        return 1

    # 2. Tests.
    # 2. 运行测试。
    if run("flutter", ["test"]) != 0:
        sys.stderr.write("flutter test failed.\n")
        return 1

    # 3. Publish dry-run (skipped for apps / private packages).
    # 3. 发布预检（应用 / 私有包可加 --no-publish 跳过）。
    if not skip_publish:
        if run("flutter", ["pub", "publish", "--dry-run"]) != 0:
            sys.stderr.write("flutter pub publish --dry-run failed.\n")
            return 1

    print("\nAll checks passed.")
    return 0


if __name__ == "__main__":
    sys.exit(main(sys.argv[1:]))

File: scripts/verify.dart
#!/usr/bin/env dart
// verify.dart — runs the local pre-push verification suite for a Flutter project.
//
// Usage:  dart run scripts/verify.dart [--no-publish]
// Runs:  flutter analyze, flutter test, and (unless skipped) flutter pub publish --dry-run.
// Exits non-zero if any step fails, so it can gate CI or a pre-push hook.

import 'dart:io';

/// Runs [cmd] with [args] in the current directory; returns its exit code.
Future<int> run(String cmd, List<String> args) async {
  print('\n> $cmd ${args.join(' ')}');
  final result = await Process.run(cmd, args,
      runInShell: true, stdout: stdout, stderr: stderr);
  return result.exitCode;
}

Future<void> main(List<String> args) async {
  final skipPublish = args.contains('--no-publish');

  // 1. Static analysis.
  if (await run('flutter', ['analyze']) != 0) {
    stderr.writeln('flutter analyze failed.');
    exit(1);
  }

  // 2. Tests.
  if (await run('flutter', ['test']) != 0) {
    stderr.writeln('flutter test failed.');
    exit(1);
  }

  // 3. Publish dry-run (skipped for apps / private packages).
  if (!skipPublish) {
    if (await run('flutter', ['pub', 'publish', '--dry-run']) != 0) {
      stderr.writeln('flutter pub publish --dry-run failed.');
      exit(1);
    }
  }

  print('\nAll checks passed.');
}
